model/src/main/java/com/flipkart/flux/domain/AuditRecord.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2012-2016, the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.flux.domain;
+
+import javax.persistence.*;
+import java.util.Date;
+
+/**
+ * <code>AuditRecord</code> represents a audit log of state machine execution.
+ * @author shyam.akirala
+ */
+
+@Entity
+public class AuditRecord {
+
+    /** Auto generated id */
+    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    /** Name of the state machine to which this audit belongs */
+    private String stateMachineName;
+
+    /** Instance id of the state machine to which this audit belongs */
+    private String stateMachineInstanceId;
+
+    /** The State identifier to which this audit belongs */
+    private Long stateId;
+
+    /** The State execution retry count */
+    private int retryAttempt;
+
+    /** The State execution status */
+    @Enumerated(EnumType.STRING)
+    private Status stateStatus;
+
+    /** Time when the State has started */
+    private Date stateStartTime;
+
+    /** Time when the State has ended */
+    private Date stateEndTime;
```

Comment:
It stores in format "2015-08-31 15:22:25.535". Will compare it with timestamp and see if it offers any additional things.